### PR TITLE
PAMv3, TMS and UUID changes

### PR DESCRIPTION
**Pull request by crimsonred**

### CODE REVIEW

File: examples/cli/cli_demo.go [hunk 1]
```diff
@@ -59,11 +59,9 @@ func connect() {
 	config.SubscribeKey = "sub-c-26a73b0a-c3f2-11e9-8b24-569e8a5c3af3" //"demo" //"sub-c-10b61350-bec7-11e9-a375-f698c1d99dce" //"demo" //
 	//config.SecretKey = //"pam"    //"demo"
 
-	//config.PublishKey = "pub-c-cdea0ef1-c571-4b72-b43f-ff1dc8aa4c5d"
-	//config.SubscribeKey = "sub-c-4757f09c-c3f2-11e9-9d00-8a58a5558306"
-	//config.SecretKey = "sec-c-YTYxNzVjYzctNDY2MS00N2NmLTg2NjYtNGRlNWY1NjMxMDBm"
-
-	//config.AuthKey = "akey"
+	config.PublishKey = "pub-c-cdea0ef1-c571-4b72-b43f-ff1dc8aa4c5d"
```
Comment (davidnub) [inline examples/cli/cli_demo.go]: Should these be removed from the `examples`?

File: examples/cli/cli_demo.go [hunk 2]
```diff
@@ -550,6 +567,35 @@ func readCommand(cmd string) {
 	}
 }
 
+func setToken(args []string) {
+	pn.SetToken(args[0])
+}
+
+//p0F2AkF0Gl2AX-JDdHRsCkNyZXOkRGNoYW6gQ2dycKBDdXNyoWl1LTMzNTIwNTUPQ3NwY6Fpcy0xNzA3OTgzGB9DcGF0pERjaGFuoENncnCgQ3VzcqBDc3BjoERtZXRhoENzaWdYINqGs2EyEMHPZrp6znVqTBzXNBAD_31hUH3JuUSWE2A6
```
Comment (davidnub) [inline examples/cli/cli_demo.go]: Should these be removed?

File: examples/snippets/onboarding.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import (
+	"fmt"
+	pubnub "github.com/pubnub/go"
+)
+
+func main() {
+
+	// Config
+	config := pubnub.NewConfig()
+	config.PublishKey = "pub-c-3ed95c83-12e6-4cda-9d69-c47ba2abb57e"   //"demo"   //"demo"
```
Comment (davidnub) [inline examples/snippets/onboarding.go]: Should we remove these from `examples` ?

File: examples/cli/cli_demo.go (hunk 1 above)
Comment (crimsonred) [reply examples/cli/cli_demo.go]: these keys will be disabled, once the release is done. These keys are used in tests as well.

File: examples/cli/cli_demo.go (hunk 2 above)
Comment (crimsonred) [reply examples/cli/cli_demo.go]: A token as a example in SetToken method. TTL is already expired.

File: examples/snippets/onboarding.go (hunk 3 above)
Comment (crimsonred) [reply examples/snippets/onboarding.go]: same keys, will be disbled

File: objects_create_space_test.go [hunk 4]
```diff
@@ -95,7 +95,7 @@ func TestCreateSpaceResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":{"id":"id2","name":"name","description":"desc","custom":{"a":"b"},"created":"2019-08-19T15:05:51.493894Z","updated":"2019-08-19T15:05:51.493894Z","eTag":"Aee9zsKNndXlHw"}}`)
 
 	r, _, err := newPNCreateSpaceResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_create_space_test.go]: Should this be commented out or just removed?

File: objects_create_user_test.go [hunk 5]
```diff
@@ -97,7 +97,7 @@ func TestCreateUserResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":{"id":"id2","name":"name","externalId":"extid","profileUrl":"purl","email":"email","custom":{"a":"b","c":"d"},"created":"2019-08-19T14:44:54.837392Z","updated":"2019-08-19T14:44:54.837392Z","eTag":"AbyT4v2p6K7fpQE"}}`)
 
 	r, _, err := newPNCreateUserResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_create_user_test.go]: Commented out or removed?

File: objects_delete_space_test.go [hunk 6]
```diff
@@ -78,7 +78,7 @@ func TestDeleteSpaceResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":null}`)
 
 	r, _, err := newPNDeleteSpaceResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_delete_space_test.go]: Commented out or removed?

File: objects_delete_user_test.go [hunk 7]
```diff
@@ -78,7 +78,7 @@ func TestDeleteUserResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":null}`)
 
 	r, _, err := newPNDeleteUserResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_delete_user_test.go]: Comment out or remove?

File: objects_get_members_test.go [hunk 8]
```diff
@@ -101,7 +101,7 @@ func TestGetMembersResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":[{"id":"id0","custom":{"a3":"b3","c3":"d3"},"user":{"id":"id0","name":"name","externalId":"extid","profileUrl":"purl","email":"email","custom":{"a":"b","c":"d"},"created":"2019-08-20T13:26:19.140324Z","updated":"2019-08-20T13:26:19.140324Z","eTag":"AbyT4v2p6K7fpQE"},"created":"2019-08-20T13:26:24.07832Z","updated":"2019-08-20T13:26:24.07832Z","eTag":"AamrnoXdpdmzjwE"}],"totalCount":1,"next":"MQ","prev":"NQ"}`)
 
 	r, _, err := newPNGetMembersResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_get_members_test.go]: Comment out or remove?

File: objects_get_memberships_test.go [hunk 9]
```diff
@@ -101,7 +101,7 @@ func TestGetMembershipsResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":[{"id":"id0","custom":{"a3":"b3","c3":"d3"},"space":{"id":"id0","name":"name","description":"desc","custom":{"a":"b"},"created":"2019-08-20T13:26:08.341297Z","updated":"2019-08-20T13:26:08.341297Z","eTag":"Aee9zsKNndXlHw"},"created":"2019-08-20T13:26:24.07832Z","updated":"2019-08-20T13:26:24.07832Z","eTag":"AamrnoXdpdmzjwE"}],"totalCount":1,"next":"MQ", "prev":"NQ"}`)
 
 	r, _, err := newPNGetMembershipsResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_get_memberships_test.go]: Comment out or remove?

File: objects_get_space_test.go [hunk 10]
```diff
@@ -90,7 +90,7 @@ func TestGetSpaceResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":{"id":"id0","name":"name","description":"desc","custom":{"a":"b"},"created":"2019-08-20T13:26:08.341297Z","updated":"2019-08-20T13:26:08.341297Z","eTag":"Aee9zsKNndXlHw"}}`)
 
 	r, _, err := newPNGetSpaceResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_get_space_test.go]: Comment out or remove?

File: objects_get_spaces_test.go [hunk 11]
```diff
@@ -99,7 +99,7 @@ func TestGetSpacesResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":[{"id":"id0","name":"name","description":"desc","custom":{"a":"b"},"created":"2019-08-20T13:26:08.341297Z","updated":"2019-08-20T13:26:08.341297Z","eTag":"Aee9zsKNndXlHw"},{"id":"id01","name":"name","description":"desc","custom":{"a":"b"},"created":"2019-08-20T14:44:52.799969Z","updated":"2019-08-20T14:44:52.799969Z","eTag":"Aee9zsKNndXlHw"}],"totalCount":2,"next":"Mg","prev":"Nd"}`)
 
 	r, _, err := newPNGetSpacesResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_get_spaces_test.go]: Comment out or remove?

File: objects_get_user_test.go [hunk 12]
```diff
@@ -90,7 +90,7 @@ func TestGetUserResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":{"id":"id0","name":"name","externalId":"extid","profileUrl":"purl","email":"email","custom":{"a":"b","c":"d"},"created":"2019-08-20T13:26:19.140324Z","updated":"2019-08-20T13:26:19.140324Z","eTag":"AbyT4v2p6K7fpQE"}}`)
 
 	r, _, err := newPNGetUserResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_get_user_test.go]: Comment out or remove?

File: objects_get_users_test.go [hunk 13]
```diff
@@ -99,7 +99,7 @@ func TestGetUsersResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":[{"id":"id2","name":"name","externalId":"extid","profileUrl":"purl","email":"email","custom":{"a":"b","c":"d"},"created":"2019-08-19T14:44:54.837392Z","updated":"2019-08-19T14:44:54.837392Z","eTag":"AbyT4v2p6K7fpQE"},{"id":"id0","name":"name","externalId":"extid","profileUrl":"purl","email":"email","custom":{"a":"b","c":"d"},"created":"2019-08-20T13:26:19.140324Z","updated":"2019-08-20T13:26:19.140324Z","eTag":"AbyT4v2p6K7fpQE"}],"totalCount":2,"next":"Mg","prev":"Nd"}`)
 
 	r, _, err := newPNGetUsersResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_get_users_test.go]: Comment out or remove?

File: objects_manage_members_test.go [hunk 14]
```diff
@@ -142,7 +142,7 @@ func TestManageMembersResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":[{"id":"userid4","custom":{"a1":"b1","c1":"d1"},"user":{"id":"userid4","name":"userid4name","externalId":"extid","profileUrl":"purl","email":"email","custom":{"a":"b","c":"d"},"created":"2019-08-23T10:36:27.083453Z","updated":"2019-08-23T10:36:27.083453Z","eTag":"AbuLvdnC9JnYEA"},"created":"2019-08-23T10:41:35.503214Z","updated":"2019-08-23T10:41:35.503214Z","eTag":"AZK3l4nQsrWG9gE"}],"totalCount":1,"next":"MQ"}`)
 
 	r, _, err := newPNManageMembersResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_manage_members_test.go]: Comment out or remove?

File: objects_manage_memberships_test.go [hunk 15]
```diff
@@ -143,7 +143,7 @@ func TestManageMembershipsResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":[{"id":"spaceid3","custom":{"a3":"b3","c3":"d3"},"space":{"id":"spaceid3","name":"spaceid3name","description":"spaceid3desc","custom":{"a":"b"},"created":"2019-08-23T10:34:43.985248Z","updated":"2019-08-23T10:34:43.985248Z","eTag":"Aazjn7vC3oDDYw"},"created":"2019-08-23T10:41:17.156491Z","updated":"2019-08-23T10:41:17.156491Z","eTag":"AamrnoXdpdmzjwE"}],"totalCount":1,"next":"MQ"}`)
 
 	r, _, err := newPNManageMembershipsResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_manage_memberships_test.go]: Comment out or remove?

File: objects_update_space_test.go [hunk 16]
```diff
@@ -95,7 +95,7 @@ func TestUpdateSpaceResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":{"id":"id0","name":"name","description":"desc","custom":{"a":"b","c":"d"},"created":"2019-08-20T13:26:08.341297Z","updated":"2019-08-20T14:48:11.675743Z","eTag":"AYKH2s7ZlYKoJA"}}`)
 
 	r, _, err := newPNUpdateSpaceResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_update_space_test.go]: Comment out or remove?

File: objects_update_user_test.go [hunk 17]
```diff
@@ -97,7 +97,7 @@ func TestUpdateUserResponseValuePass(t *testing.T) {
 	jsonBytes := []byte(`{"status":200,"data":{"id":"id0","name":"name","externalId":"extid","profileUrl":"purl","email":"email","custom":{"a":"b","c":"d"},"created":"2019-08-20T13:26:19.140324Z","updated":"2019-08-20T13:26:19.140324Z","eTag":"AbyT4v2p6K7fpQE"}}`)
 
 	r, _, err := newPNUpdateUserResponse(jsonBytes, opts, StatusResponse{})
-	assert.Equal(200, r.Status)
+	//assert.Equal(200, r.status)
```
Comment (davidnub) [inline objects_update_user_test.go]: Comment out or remove?

File: tests/e2e/message_counts_test.go [hunk 18]
```diff
@@ -4,7 +4,7 @@ import (
 	"fmt"
 	//"log"
 	//"os"
-	"strconv"
+	//"strconv"
```
Comment (davidnub) [inline tests/e2e/message_counts_test.go]: Why not just remove the commented out stuff?

File: examples/cli/cli_demo.go (hunk 1 above)
Comment (crimsonred) [reply examples/cli/cli_demo.go]: Removed

File: examples/cli/cli_demo.go (hunk 2 above)
Comment (crimsonred) [reply examples/cli/cli_demo.go]: Removed

File: examples/snippets/onboarding.go (hunk 3 above)
Comment (crimsonred) [reply examples/snippets/onboarding.go]: Removed

File: objects_create_space_test.go (hunk 4 above)
Comment (crimsonred) [reply objects_create_space_test.go]: Removed

File: objects_create_user_test.go (hunk 5 above)
Comment (crimsonred) [reply objects_create_user_test.go]: Removed

File: objects_delete_space_test.go (hunk 6 above)
Comment (crimsonred) [reply objects_delete_space_test.go]: Removed

File: objects_delete_user_test.go (hunk 7 above)
Comment (crimsonred) [reply objects_delete_user_test.go]: Removed

File: objects_get_members_test.go (hunk 8 above)
Comment (crimsonred) [reply objects_get_members_test.go]: removed

File: objects_get_memberships_test.go (hunk 9 above)
Comment (crimsonred) [reply objects_get_memberships_test.go]: removed

File: objects_get_space_test.go (hunk 10 above)
Comment (crimsonred) [reply objects_get_space_test.go]: removed

File: objects_get_spaces_test.go (hunk 11 above)
Comment (crimsonred) [reply objects_get_spaces_test.go]: removed

File: objects_get_user_test.go (hunk 12 above)
Comment (crimsonred) [reply objects_get_user_test.go]: removed

File: objects_get_users_test.go (hunk 13 above)
Comment (crimsonred) [reply objects_get_users_test.go]: removed

File: objects_manage_members_test.go (hunk 14 above)
Comment (crimsonred) [reply objects_manage_members_test.go]: removed

File: objects_manage_memberships_test.go (hunk 15 above)
Comment (crimsonred) [reply objects_manage_memberships_test.go]: removed

File: objects_update_space_test.go (hunk 16 above)
Comment (crimsonred) [reply objects_update_space_test.go]: removed

File: objects_update_user_test.go (hunk 17 above)
Comment (crimsonred) [reply objects_update_user_test.go]: removed

File: tests/e2e/message_counts_test.go (hunk 18 above)
Comment (crimsonred) [reply tests/e2e/message_counts_test.go]: removed

**Pull request merged**